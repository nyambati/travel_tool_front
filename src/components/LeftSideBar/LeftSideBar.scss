.LeftSideBar {
  margin: 0px;
  width: 240px;
  height: calc(100vh - 78px);
  border-right: 1px solid #E4E4E4;
  font-family: 'DIN Pro';

  &__fixed_wrapper {
    position: fixed;
    top: 128px;
    width: 240px;
    height: calc(100vh - 128px);
  }
}
