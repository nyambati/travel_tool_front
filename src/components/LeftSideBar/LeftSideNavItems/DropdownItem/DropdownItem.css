.DropdownItem {
  display: flex;
  height: 21px;
  font-size: 16px;
  font-weight: 500;
  color: #4F4F4F;
  margin-bottom: 25px;
}

.DropdownItem.active {
  color: #3359DB;
}

.DropdownItem:not([class~="active"]):hover {
  color: black;
}
