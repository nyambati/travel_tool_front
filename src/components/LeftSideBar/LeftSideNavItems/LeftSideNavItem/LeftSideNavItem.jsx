import React, { Fragment } from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import DropdownIcon from "../../../../images/icons/dropdown";
import "./LeftSideNavItem.scss";

export class LeftSideNavItem extends React.PureComponent {
  static propTypes = {
    text: PropTypes.string.isRequired,
    link_to: PropTypes.string.isRequired,
    iconSrc: PropTypes.func.isRequired,
    isDropdown: PropTypes.bool,
    children: PropTypes.func,
    match: PropTypes.object.isRequired
  }

  static defaultProps = {
    children: () => {},
    isDropdown: false
  }

  constructor(props) {
    super(props);
    this.state = {
      dropdownActive: false,
    };
  }

  getDropdownElements = (children, dropdownStatus, linkIsActive) => {
    const dropdownItems = (
      <div className={`LeftSideNavItem__dropdownContent ${dropdownStatus}`}>
        {children}
      </div>
    );

    const dropdownIcon = (
      <div
        className={`dropdownIcon dropdownIcon--${dropdownStatus}`}
        style={{position: "absolute", right: "30px", top: "17px"}}>
        <DropdownIcon active={linkIsActive} />
      </div>
    );

    return {
      items: dropdownItems,
      icon: dropdownIcon
    };
  }


  toggleDropdown = () => {
    const { dropdownActive } = this.state;
    this.setState({dropdownActive: !dropdownActive});
  }

  render() {
    const {text, link_to, iconSrc, isDropdown, children, match } = this.props;
    const linkIsActive = match.path.startsWith(link_to);
    const LeftIcon = iconSrc;
    const { dropdownActive } = this.state;
    let dropdownStatus = dropdownActive? 'active': 'inactive';
    let dropdownElements;

    if (isDropdown)
      dropdownElements = this.getDropdownElements(children, dropdownStatus, linkIsActive);
      
    return (
      <Fragment>
        <NavLink onClick={this.toggleDropdown} to={link_to} className="LeftSideNavItem">
          <div className="LeftSideNavItem__leftIcon">
            <LeftIcon active={linkIsActive} />
          </div>
          <div className="LeftSideNavItem__Nav-text">
            {text}
          </div>
          { dropdownElements? dropdownElements.icon: null}
        </NavLink>
        {dropdownElements? dropdownElements.items: null}
      </Fragment>
    );
  }
}

export default withRouter(LeftSideNavItem);
