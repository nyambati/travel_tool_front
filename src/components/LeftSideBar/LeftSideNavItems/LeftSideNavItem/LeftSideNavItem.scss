.LeftSideNavItem {
  position: relative;
  display: flex;
  flex-flow: row;
  height: 48px;
  align-items: center;
  border-left: 3px solid transparent;
  padding-left: 47px;
  font-size: 18px;
  font-weight: 500;
  color: #999999;

  &__Nav-text {
    position: absolute;
    left: 85px;
  }

  &__leftIcon {
    position: absolute;
    top: 13px;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 24px;
    width: 24px;
  }

  .dropdownIcon {
    transition: transform .3s ease-out;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 18px;
    width: 18px;

    &--inactive {
      transform: rotate(180deg);
    }
  }


  &__dropdownContent {
    display: flex;
    flex-flow: column;
    justify-content:flex-start;
    transition: all .5s ease-out;
    padding-left: 88px;
    padding-top: 0px;
    overflow-y: hidden;
  }

  &__dropdownContent.active {
    height: 116.5px;
    padding-top: 25px;
  }

  &__dropdownContent.inactive {
    height: 0px;
    overflow-y: hidden;
  }
}

.LeftSideNavItem:not([class~="active"]):hover {
  color: #4F4F4F;
}

.LeftSideNavItem.active {
  color: #3359DB;
  border-color: #3359DB;
  background-color: #EEF0F6;
  border-right: 1px solid #E4E4E4;
}
