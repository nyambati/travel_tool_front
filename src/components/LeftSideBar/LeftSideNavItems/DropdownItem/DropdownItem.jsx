import React from 'react';
import { NavLink } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import "./DropdownItem.css";

const DropdownItem = (props) => {
  const { children, link_to } = props;
  return (
    <NavLink to={link_to} className="DropdownItem">
      {children}
    </NavLink>
  );
};

DropdownItem.propTypes = {
  link_to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

export default DropdownItem;
