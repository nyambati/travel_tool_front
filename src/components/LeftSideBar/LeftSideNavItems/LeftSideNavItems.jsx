import React, { Fragment } from "react";
import LeftSidebarNavItem from "./LeftSideNavItem/LeftSideNavItem";
import DropdownItem from "./DropdownItem/DropdownItem";
import BookmarkIcon from "../../../images/icons/bookmark";
import SettingsIcon from "../../../images/icons/settings";


const LeftSideNavItems = () => (
  <Fragment>
    <LeftSidebarNavItem isDropdown iconSrc={BookmarkIcon} link_to="/requests" text="Requests">
      <DropdownItem link_to="/requests/my-requests">
        My Requests
      </DropdownItem>
      <DropdownItem link_to="/requests/my-approvals">
        My Approvals
      </DropdownItem>
    </LeftSidebarNavItem>
    <LeftSidebarNavItem iconSrc={SettingsIcon} link_to="/settings" text="Settings" />
  </Fragment>
);

export default LeftSideNavItems;
