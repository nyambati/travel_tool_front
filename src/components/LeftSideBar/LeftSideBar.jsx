import React from "react";
import LeftSideNavItems from "./LeftSideNavItems/LeftSideNavItems";
import "./LeftSideBar.scss";

const LeftSideBar = () => (
  <div className="LeftSideBar mdl-cell mdl-cell--2-col">
    <div className="LeftSideBar__fixed_wrapper">
      <div style={{overflowY: "scroll", height: "100%"}}>
        <LeftSideNavItems />
      </div>
    </div>
  </div>
);

export default LeftSideBar;
