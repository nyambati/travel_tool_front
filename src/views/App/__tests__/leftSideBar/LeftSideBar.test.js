import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import LeftSideBar from '../../../../components/LeftSideBar/LeftSideBar';
import LeftSideNavItems from '../../../../components/LeftSideBar/LeftSideNavItems/LeftSideNavItems';
import { LeftSideNavItem } from '../../../../components/LeftSideBar/LeftSideNavItems/LeftSideNavItem/LeftSideNavItem';
import DropdownItem from '../../../../components/LeftSideBar/LeftSideNavItems/DropdownItem/DropdownItem';
import BookmarkIcon from '../../../../images/icons/bookmark';
import DropdownIcon from '../../../../images/icons/dropdown';
import SettingsIcon from '../../../../images/icons/settings';


describe('<LeftSideBar /> tests', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<LeftSideBar />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('<LeftSideNavItems /> tests', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<LeftSideNavItems />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('it renders two dropdown items', () => {
      const dropDownItems = wrapper.find('DropdownItem');
      expect(dropDownItems).toHaveLength(2);
    });

    describe('<LeftSideNavItem tests', () => {
      it('renders the LeftSideNavItem', () => {
        const match = {path: '/test-link'};
        const iconSrc = () => {};
        const wrapper = shallow(
          <LeftSideNavItem
            text="Test link"
            match={match}
            iconSrc={iconSrc}
            link_to="/test-link" />
        );
        expect(wrapper).toMatchSnapshot();
      });

      describe('Dropdown Navigation item', () => {
        let match, iconSrc, wrapper;

        beforeEach(() => {
          match = {path: '/test-link'};
          iconSrc = () => {};
          wrapper = shallow(
            <LeftSideNavItem
              text="Test link"
              isDropdown
              iconSrc={iconSrc}
              match={match}
              link_to="/test-link" />
          );
        });

        it('renders as dropdown', () => {
          expect(wrapper).toMatchSnapshot();
        });

        it('toggles dropdown', () => {
            const expectedStateOnClick = { dropdownActive: true };

            wrapper.find('NavLink').simulate('click');
            const stateOnClick = wrapper
              .instance()
              .state;
            expect(stateOnClick).toEqual(expectedStateOnClick);
        });
      });
    });

    describe('<DropdownItem />', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <DropdownItem link_to="/test-link/test-endpoint">
            Test Link
          </DropdownItem>
        );

        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('SVG Icons', () => {
      it('renders icons', () => {
        const bookmarkIconWrapper = shallow(<BookmarkIcon active />);
        const dropdownIconWrapper = shallow(<DropdownIcon active />);
        const settingsIconWrapper = shallow(<SettingsIcon active />);
        expect(bookmarkIconWrapper).toMatchSnapshot();
        expect(dropdownIconWrapper).toMatchSnapshot();
        expect(settingsIconWrapper).toMatchSnapshot();
      });
    });
  });
});
