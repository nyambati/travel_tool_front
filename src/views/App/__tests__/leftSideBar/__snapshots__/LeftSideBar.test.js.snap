// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests <DropdownItem /> renders correctly 1`] = `
<NavLink
  activeClassName="active"
  aria-current="page"
  className="DropdownItem"
  to="/test-link/test-endpoint"
>
  Test Link
</NavLink>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests <LeftSideNavItem tests Dropdown Navigation item renders as dropdown 1`] = `
<React.Fragment>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="LeftSideNavItem"
    onClick={[Function]}
    to="/test-link"
  >
    <div
      className="LeftSideNavItem__leftIcon"
    >
      <iconSrc
        active={true}
      />
    </div>
    <div
      className="LeftSideNavItem__Nav-text"
    >
      Test link
    </div>
    <div
      className="dropdownIcon dropdownIcon--inactive"
      style={
        Object {
          "position": "absolute",
          "right": "30px",
          "top": "17px",
        }
      }
    >
      <DropdownIcon
        active={true}
      />
    </div>
  </NavLink>
  <div
    className="LeftSideNavItem__dropdownContent inactive"
  />
</React.Fragment>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests <LeftSideNavItem tests renders the LeftSideNavItem 1`] = `
<React.Fragment>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="LeftSideNavItem"
    onClick={[Function]}
    to="/test-link"
  >
    <div
      className="LeftSideNavItem__leftIcon"
    >
      <iconSrc
        active={true}
      />
    </div>
    <div
      className="LeftSideNavItem__Nav-text"
    >
      Test link
    </div>
  </NavLink>
</React.Fragment>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests SVG Icons renders icons 1`] = `
<svg
  height="18px"
  version="1.1"
  viewBox="0 0 14 18"
  width="14px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <title>
    7D627ECF-F811-4218-B0FD-305F290B9E25
  </title>
  <defs>
    <polygon
      id="path-1"
      points="0 12.97 5 10.82 10 12.97 10 0 0 0"
    />
    <path
      d="M12,0 L2,0 C0.9,0 0,0.9 0,2 L0,18 L7,15 L14,18 L14,2 C14,0.9 13.1,0 12,0 Z M12,14.97 L7,12.83 L2,14.97 L2,2 L12,2 L12,14.97 Z"
      id="path-3"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    id="Web-App"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Requests"
      transform="translate(-55.000000, -144.000000)"
    >
      <g
        id="Group-5"
        transform="translate(0.000000, 128.000000)"
      >
        <g
          id="icon/action/bookmark_24px"
          transform="translate(50.000000, 13.000000)"
        >
          <g
            id="↳-Light-Color"
            transform="translate(7.000000, 5.000000)"
          >
            <mask
              fill="white"
              id="bookmark-mask-2"
            >
              <use
                xlinkHref="#path-1"
              />
            </mask>
            <g
              fillRule="nonzero"
              id="bookmark-mask"
              opacity="0.3"
            />
            <g
              fill="#FFFFFF"
              id="✱-/-Color-/-Fill-/-transparent"
              mask="url(#bookmark-mask-2)"
              opacity="0"
            >
              <rect
                height="101"
                id="Rectangle"
                width="101"
                x="0"
                y="0"
              />
            </g>
          </g>
          <g
            id="↳-Dark-Color"
            transform="translate(5.000000, 3.000000)"
          >
            <mask
              fill="white"
              id="bookmark-mask-4"
            >
              <use
                xlinkHref="#path-3"
              />
            </mask>
            <g
              fillRule="nonzero"
              id="bookmark-mask"
            />
            <g
              fill="#3359DB"
              id="✱-/-Color-/-Border-/-Active"
              mask="url(#bookmark-mask-4)"
            >
              <rect
                height="101"
                id="Rectangle-13"
                width="101"
                x="0"
                y="0"
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests SVG Icons renders icons 2`] = `
<svg
  height="6px"
  version="1.1"
  viewBox="0 0 10 6"
  width="10px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsHlink="http://www.w3.org/1999/xlink"
>
  <title>
    0FE89DFF-A8FB-4D5C-95C0-4A3B10101A32
  </title>
  <desc>
    Created with sketchtool.
  </desc>
  <defs>
    <polygon
      id="dropdown-path-1"
      points="9 6.22125 4.5 10.72125 5.5575 11.77875 9 8.34375 12.4425 11.77875 13.5 10.72125"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    id="Web-App"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Requests"
      transform="translate(-196.000000, -151.000000)"
    >
      <g
        id="Group-5"
        transform="translate(0.000000, 128.000000)"
      >
        <g
          id="icon/navigation/expand_less_24px"
          transform="translate(192.000000, 17.000000)"
        >
          <mask
            fill="white"
            id="dropdown-mask-2"
          >
            <use
              xlinkHref="#dropdown-path-1"
            />
          </mask>
          <g
            fillRule="nonzero"
          />
          <g
            fill="#3359DB"
            fillRule="evenodd"
            id="✱-/-Color-/-Border-/-Active"
            mask="url(#dropdown-mask-2)"
          >
            <rect
              height="18"
              id="Rectangle-13"
              width="18"
              x="0"
              y="0"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests SVG Icons renders icons 3`] = `
<svg
  height="20px"
  version="1.1"
  viewBox="0 0 20 20"
  width="20px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <title>
    AC036E8B-A651-4E93-8335-E02BC5C33BAC
  </title>
  <desc>
    Created with sketchtool.
  </desc>
  <defs>
    <path
      d="M15.28,4.6 L14.58,3.39 L13.31,3.9 L12.25,4.33 L11.34,3.63 C10.95,3.33 10.54,3.09 10.11,2.92 L9.05,2.49 L8.89,1.36 L8.7,0 L7.3,0 L7.11,1.35 L6.95,2.48 L5.89,2.92 C5.48,3.09 5.07,3.33 4.64,3.65 L3.74,4.33 L2.69,3.91 L1.42,3.39 L0.72,4.6 L1.8,5.44 L2.69,6.14 L2.55,7.27 C2.52,7.57 2.5,7.8 2.5,8 C2.5,8.2 2.52,8.43 2.55,8.73 L2.69,9.86 L1.8,10.56 L0.72,11.4 L1.42,12.61 L2.69,12.1 L3.75,11.67 L4.66,12.37 C5.05,12.67 5.46,12.91 5.89,13.08 L6.95,13.51 L7.11,14.64 L7.3,16 L8.69,16 L8.88,14.65 L9.04,13.52 L10.1,13.09 C10.51,12.92 10.92,12.68 11.35,12.36 L12.25,11.68 L13.29,12.1 L14.56,12.61 L15.26,11.4 L14.18,10.56 L13.29,9.86 L13.43,8.73 C13.47,8.42 13.48,8.21 13.48,8 C13.48,7.79 13.46,7.57 13.43,7.27 L13.29,6.14 L14.18,5.44 L15.28,4.6 Z M8,12 C5.79,12 4,10.21 4,8 C4,5.79 5.79,4 8,4 C10.21,4 12,5.79 12,8 C12,10.21 10.21,12 8,12 Z"
      id="settings-path-1"
    />
    <path
      d="M17.43,10.98 C17.47,10.66 17.5,10.34 17.5,10 C17.5,9.66 17.47,9.34 17.43,9.02 L19.54,7.37 C19.73,7.22 19.78,6.95 19.66,6.73 L17.66,3.27 C17.57,3.11 17.4,3.02 17.22,3.02 C17.16,3.02 17.1,3.03 17.05,3.05 L14.56,4.05 C14.04,3.65 13.48,3.32 12.87,3.07 L12.49,0.42 C12.46,0.18 12.25,0 12,0 L8,0 C7.75,0 7.54,0.18 7.51,0.42 L7.13,3.07 C6.52,3.32 5.96,3.66 5.44,4.05 L2.95,3.05 C2.89,3.03 2.83,3.02 2.77,3.02 C2.6,3.02 2.43,3.11 2.34,3.27 L0.34,6.73 C0.21,6.95 0.27,7.22 0.46,7.37 L2.57,9.02 C2.53,9.34 2.5,9.67 2.5,10 C2.5,10.33 2.53,10.66 2.57,10.98 L0.46,12.63 C0.27,12.78 0.22,13.05 0.34,13.27 L2.34,16.73 C2.43,16.89 2.6,16.98 2.78,16.98 C2.84,16.98 2.9,16.97 2.95,16.95 L5.44,15.95 C5.96,16.35 6.52,16.68 7.13,16.93 L7.51,19.58 C7.54,19.82 7.75,20 8,20 L12,20 C12.25,20 12.46,19.82 12.49,19.58 L12.87,16.93 C13.48,16.68 14.04,16.34 14.56,15.95 L17.05,16.95 C17.11,16.97 17.17,16.98 17.23,16.98 C17.4,16.98 17.57,16.89 17.66,16.73 L19.66,13.27 C19.78,13.05 19.73,12.78 19.54,12.63 L17.43,10.98 Z M15.45,9.27 C15.49,9.58 15.5,9.79 15.5,10 C15.5,10.21 15.48,10.43 15.45,10.73 L15.31,11.86 L16.2,12.56 L17.28,13.4 L16.58,14.61 L15.31,14.1 L14.27,13.68 L13.37,14.36 C12.94,14.68 12.53,14.92 12.12,15.09 L11.06,15.52 L10.9,16.65 L10.7,18 L9.3,18 L9.11,16.65 L8.95,15.52 L7.89,15.09 C7.46,14.91 7.06,14.68 6.66,14.38 L5.75,13.68 L4.69,14.11 L3.42,14.62 L2.72,13.41 L3.8,12.57 L4.69,11.87 L4.55,10.74 C4.52,10.43 4.5,10.2 4.5,10 C4.5,9.8 4.52,9.57 4.55,9.27 L4.69,8.14 L3.8,7.44 L2.72,6.6 L3.42,5.39 L4.69,5.9 L5.73,6.32 L6.63,5.64 C7.06,5.32 7.47,5.08 7.88,4.91 L8.94,4.48 L9.1,3.35 L9.3,2 L10.69,2 L10.88,3.35 L11.04,4.48 L12.1,4.91 C12.53,5.09 12.93,5.32 13.33,5.62 L14.24,6.32 L15.3,5.89 L16.57,5.38 L17.27,6.59 L16.2,7.44 L15.31,8.14 L15.45,9.27 Z"
      id="settings-path-3"
    />
    <path
      d="M4,0 C1.79,0 0,1.79 0,4 C0,6.21 1.79,8 4,8 C6.21,8 8,6.21 8,4 C8,1.79 6.21,0 4,0 Z M4,6 C2.9,6 2,5.1 2,4 C2,2.9 2.9,2 4,2 C5.1,2 6,2.9 6,4 C6,5.1 5.1,6 4,6 Z"
      id="settings-path-5"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    id="Web-App"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Requests"
      transform="translate(-52.000000, -308.000000)"
    >
      <g
        id="Group-4-Copy-2"
        transform="translate(50.000000, 306.000000)"
      >
        <g
          id="icon/action/settings_24px"
        >
          <g
            id="↳-Light-Color"
            transform="translate(4.000000, 4.000000)"
          >
            <mask
              fill="white"
              id="settings-mask-2"
            >
              <use
                xlinkHref="#settings-path-1"
              />
            </mask>
            <g
              fillRule="nonzero"
              id="settings-mask"
              opacity="0.3"
            />
            <g
              fill="#FFFFFF"
              id="✱-/-Color-/-Fill-/-transparent"
              mask="url(#settings-mask-2)"
              opacity="0"
            >
              <rect
                height="101"
                id="Rectangle"
                width="101"
                x="0"
                y="0"
              />
            </g>
          </g>
          <g
            id="↳-Dark-Color"
            transform="translate(2.000000, 2.000000)"
          >
            <mask
              fill="white"
              id="settings-mask-4"
            >
              <use
                xlinkHref="#settings-path-3"
              />
            </mask>
            <g
              fillRule="nonzero"
              id="settings-mask"
            />
            <g
              fill="#3359DB"
              id="✱-/-Color-/-Border-/-Inactive"
              mask="url(#settings-mask-4)"
            >
              <rect
                height="101"
                id="Rectangle"
                width="101"
                x="0"
                y="0"
              />
            </g>
          </g>
          <g
            id="↳-Dark-Color"
            transform="translate(8.000000, 8.000000)"
          >
            <mask
              fill="white"
              id="settings-mask-6"
            >
              <use
                xlinkHref="#settings-path-5"
              />
            </mask>
            <g
              fillRule="nonzero"
              id="settings-mask"
            />
            <g
              fill="#3359DB"
              id="✱-/-Color-/-Border-/-Inactive"
              mask="url(#settings-mask-6)"
            >
              <rect
                height="101"
                id="Rectangle"
                width="101"
                x="0"
                y="0"
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`<LeftSideBar /> tests <LeftSideNavItems /> tests renders correctly 1`] = `
<React.Fragment>
  <withRouter(LeftSideNavItem)
    iconSrc={[Function]}
    isDropdown={true}
    link_to="/requests"
    text="Requests"
  >
    <DropdownItem
      link_to="/requests/my-requests"
    >
      My Requests
    </DropdownItem>
    <DropdownItem
      link_to="/requests/my-approvals"
    >
      My Approvals
    </DropdownItem>
  </withRouter(LeftSideNavItem)>
  <withRouter(LeftSideNavItem)
    iconSrc={[Function]}
    link_to="/settings"
    text="Settings"
  />
</React.Fragment>
`;

exports[`<LeftSideBar /> tests renders correctly 1`] = `
<div
  className="LeftSideBar mdl-cell mdl-cell--2-col"
>
  <div
    className="LeftSideBar__fixed_wrapper"
  >
    <div
      style={
        Object {
          "height": "100%",
          "overflowY": "scroll",
        }
      }
    >
      <LeftSideNavItems />
    </div>
  </div>
</div>
`;
